tests/checkers/unittest_refactoring.py

```diff
@@ -32,3 +32,17 @@ def test_process_tokens() -> None:
         with pytest.raises(SystemExit) as cm:
             Run([os.path.join(REGR_DATA, "very_long_line.py")], reporter=TextReporter())
         assert cm.value.code == 0
+
+
+@pytest.mark.skipif(not hasattr(signal, "setitimer"), reason="Assumes POSIX signals")
+def test_issue_5724() -> None:
+    with timeout(25.0):
```

Comment:
Did you check if this indeed times out without this change?

Reply:
Yes, of course.  Locally, it was >1min on my machine.  I'm not sure, however, if it safe enough for the GH CI.

For instance, I've
```
$ python -c 'print("a = " + "\"" + "a #"*1000 + "\"")' > tests/regrtest_data/issue_5724.py
$ time pylint tests/regrtest_data/issue_5724.py --enable=missing-final-newline

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)


real    1m35.922s
user    1m35.414s
sys     0m0.196s
```
vs hours for the main branch (this example is more close to #5724).  Maybe this is better?

Reply:
I think this is fine, no reason to make this test take more than a minute and I also think it is unlikely that we will ever regress here. Thanks for the additional details!